src/main/resources/assets/tconstruct/book/puny_smelting/en_us/upgrades/general.json

```diff
@@ -3,6 +3,10 @@
   "text": [
     {
       "text": "Upgrades are modifiers added to tools using upgrade recipes. Each tool has a limited number of upgrade slots to start, though the number of slots can be increased in a few ways."
+    },
+	{
+      "text": "Many of the upgrades listed in general, weapon, bonus damage, and general work on armor as well!",
```

Comment:
Weapon/bonus damage only work on chestplates for the most part, which is tied to the unarmed modifier. I just had a special blurb in the encyclopedia for that. Let me know if you see any weapon modifiers that are not tied to unarmed attack.

Also said general twice.